rzup/src/cmd/toolchain.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::Subcommand;
+
+use crate::toolchain::{build::BuildToolchain, install::InstallToolchain, repo::ToolchainRepo};
+
+use super::show::print_installed_toolchains;
+
+#[derive(Debug, Subcommand)]
+pub enum ToolchainSubcmd {
+    /// List installed toolchains
+    List {
+        #[arg(short, long)]
+        verbose: bool,
+    },
+
+    /// Install or update a given toolchain
+    Install {
+        #[command(subcommand)]
+        subcmd: ToolchainInstallSubcmd,
+    },
+
+    /// Build the toolchain
+    Build(BuildToolchain),
+
+    /// Update toolchain
+    Update {
+        #[command(subcommand)]
+        subcmd: ToolchainInstallSubcmd,
+    },
+}
+
+#[derive(Debug, Subcommand)]
+pub enum ToolchainInstallSubcmd {
+    /// Install Rust toolchain
+    Rust {
+        /// Toolchain version (i.e. 'latest' or a specific version)
+        #[arg(required = false)]
+        toolchain: Option<String>,
```

Comment:
Perhaps the error message for this could indicate what this is looking for? I ran `rzup toolchain install rust 0.21.0` and the follow error message might not be intuitive that it's looking for a git ref/tag:
```
Error during Rust toolchain installation: Failed to fetch release info

Caused by:
    HTTP status client error (404 Not Found) for url (https://api.github.com/repos/risc0/rust/releases/tags/0.21.0)
```

Ideally this links to versions we recommend?

Reply:
This is where our naming semantics gets a little hairy. `0.21.0` is a specific version of the `cargo-risczero` extension, not necessarily the toolchain. Toolchain versions are tagged by date: https://github.com/risc0/rust/releases

So the correct toolchain install command would be something like `rzup toolchain install rust v2024-04-22.0` :/

Reply:
Right, guess I was assuming a superset of what this command did. Perhaps just making clear what this version is would be helpful